internal/schema/usage_data.go

```diff
@@ -149,8 +151,64 @@ func (u *UsageData) CalcEstimationSummary() map[string]bool {
 	return estimationMap
 }
 
-func NewUsageMap(m map[string]interface{}) map[string]*UsageData {
-	usageMap := make(map[string]*UsageData)
+// UsageMap is a map of address to UsageData built from a usage file.
+// UsageMap is a standalone type so that we can do more involved matching functionality.
+type UsageMap map[string]*UsageData
+
+// Get returns UsageData for a given resource address.
+func (usage UsageMap) Get(address string) *UsageData {
+	var data *UsageData
+
+	parsedAddress, err := addressParser.NewAddress(address)
+	if err == nil {
+		val, ok := usage[parsedAddress.ResourceSpec.Type]
+		if ok {
+			data = val
+		}
+	}
+
+	if ud := usage[address]; ud != nil {
+		if data != nil {
+			MergeAttributes(data, ud)
+		}
+
+		return ud
+	}
+
+	if strings.HasSuffix(address, "]") {
+		lastIndexOfOpenBracket := strings.LastIndex(address, "[")
+
+		if arrayUsageData := usage[fmt.Sprintf("%s[*]", address[:lastIndexOfOpenBracket])]; arrayUsageData != nil {
+			return arrayUsageData
+		}
+	}
```

Comment:
technically we don't "need" this now with the regxp matching below, however this is more performant lookup. So I'd prefer to keep it, so we catch things before moving to the matching functionality. However, I'm ok to remove as well.